### DISCUSSION TITLE
Best way to disable an Actor?

### DISCUSSION THREAD

**Discussion by alexkreidler** [comunica/comunica#812] [Q&A]
I'm trying to run a SPARQL query on a Virtuoso endpoint through a local proxy `http://localhost:8010/proxy/sparql`. 

Comunica extracts the HTML microdata using the `@comunica/bus-rdf-metadata-extract` actor, finds the SPARQL service definition, and then makes subsequent requests directly to that endpoint (`http://localhost:8890/sparql`) rather than through the proxy.

I can and probably will either change the CORS settings on Virtuoso or set a Comunica-wide proxy with the context. I also just recently remembered I can just set the [source type](https://comunica.dev/docs/query/advanced/source_types/) to sparql in JS.

But nevertheless, let's suppose I just want to disable a single or several Actors. What is the best way to do this? don't really want to run Components.js to [compile my own version](https://componentsjs.readthedocs.io/en/latest/compilation/). Is there a way to do this at runtime?

Would it be possible to add to Comunica if not?

Thanks!

<details>
<summary>HTML microdata</summary>

```html
<div xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
         xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#"
         xmlns="http://www.w3.org/1999/xhtml"
         xmlns:sd="http://www.w3.org/ns/sparql-service-description#"
         xmlns:xsd="http://www.w3.org/2001/XMLSchema#"
         style="display:none">
       <div class="description" about="" typeof="sd:Service">
          <div rel="sd:endpoint" resource="http://localhost:8890/sparql"/>
          <div rel="sd:feature"
               resource="http://www.w3.org/ns/sparql-service-description#UnionDefaultGraph"/>
          <div rel="sd:feature"
               resource="http://www.w3.org/ns/sparql-service-description#DereferencesURIs"/>
          <div rel="sd:resultFormat" resource="http://www.w3.org/ns/formats/RDF_XML"/>
          <div rel="sd:resultFormat" resource="http://www.w3.org/ns/formats/Turtle"/>
          <div rel="sd:resultFormat"
               resource="http://www.w3.org/ns/formats/SPARQL_Results_CSV"/>
          <div rel="sd:resultFormat" resource="http://www.w3.org/ns/formats/N-Triples"/>
          <div rel="sd:resultFormat" resource="http://www.w3.org/ns/formats/N3"/>
          <div rel="sd:resultFormat"
               resource="http://www.w3.org/ns/formats/SPARQL_Results_JSON"/>
          <div rel="sd:resultFormat" resource="http://www.w3.org/ns/formats/RDFa"/>
          <div rel="sd:resultFormat"
               resource="http://www.w3.org/ns/formats/SPARQL_Results_XML"/>
          <div rel="sd:supportedLanguage"
               resource="http://www.w3.org/ns/sparql-service-description#SPARQL10Query"/>
          <div rel="sd:url" resource="http://localhost:8890/sparql"/>
       </div>
    </div>
```

</details>

**Comment by rubensworks** [comment]
> I also just recently remembered I can just set the source type to sparql in JS.

Indeed, doing that should be the easiest solution.

> let's suppose I just want to disable a single or several Actors. What is the best way to do this?

The best way would be to create your own config via Components.js ([we have several guides for this](https://comunica.dev/docs/modify/)).

Via [`newEngineDynamic()`](https://comunica.dev/docs/modify/getting_started/custom_config_app/#2--creating-a-new-query-engine), you can do this at runtime without compilation. But this will come at a significant cost in startup time for doing DI at runtime. So compiling a config is definitely recommended.

You could in theory also not use the config files, and compose your engine purely via JS, but I definitely do no recommend that. This would require you to write hundreds LOC, and is very error-prone. That's exactly the reason why these config files exists :-)
